Game/RainMeadow.GameHooks.cs

```diff
@@ -29,13 +29,26 @@ private void GameHooks()
             IL.Room.LoadFromDataString += Room_LoadFromDataString;
             IL.Room.Loaded += Room_Loaded;
             On.Room.Loaded += Room_LoadedCheck;
+            On.Room.ReadyForAI += Room_ReadyForAI;
 
             On.FliesWorldAI.AddFlyToSwarmRoom += FliesWorldAI_AddFlyToSwarmRoom;
             
             // Arena specific
             On.GameSession.AddPlayer += GameSession_AddPlayer;
         }
 
+        private void Room_ReadyForAI(On.Room.orig_ReadyForAI orig, Room self)
+        {
+            if (OnlineManager.lobby != null)
+            {
+                if (!RoomSession.map.TryGetValue(self.abstractRoom, out var rs)) return;
```

Comment:
Doesn't run orig if you do it this way. This is a "should never happen" scenario though so probably doesn't matter.

Reply:
Found a better function to hook into